### ISSUE TITLE
metaServer doesn't work well for cluster resources

### ISSUE THREAD

**Issue by bluven** [kubeedge/kubeedge#5343]
**What happened**:

I write a simple program to watch nodes on edge side:
watch-node.go
```go
package main

import (
	"fmt"
	"time"

	corev1 "k8s.io/api/core/v1"
	"k8s.io/client-go/informers"
	"k8s.io/client-go/kubernetes"
	toolscache "k8s.io/client-go/tools/cache"
	"k8s.io/client-go/tools/clientcmd"
	"k8s.io/klog/v2"
)

var syncPeriod = 24 * time.Hour

func main() {

	cfg, err := clientcmd.BuildConfigFromFlags("", "./kubeconfig")
	if err != nil {
		panic(err)
	}

	clientset := kubernetes.NewForConfigOrDie(cfg)
	kubeInformerFactory := informers.NewSharedInformerFactory(clientset, syncPeriod)
	nodeInformer := kubeInformerFactory.Core().V1().Nodes().Informer()
	nodeInformer.AddEventHandlerWithResyncPeriod(
		toolscache.ResourceEventHandlerFuncs{
			AddFunc:    onNodeAdd,
			UpdateFunc: OnNodeUpdate,
			DeleteFunc: OnNodeDelete,
		},
		syncPeriod,
	)

	stopChan := make(chan struct{})

	go func() {
		klog.InfoS("Starting node controller")

		if !toolscache.WaitForNamedCacheSync("node", stopChan, nodeInformer.HasSynced) {
			return
		}
	}()

	kubeInformerFactory.Start(stopChan)

	select {}
}

func onNodeAdd(obj interface{}) {
	displayNode("add", obj)
}

func OnNodeUpdate(oldObj, newObj interface{}) {
	displayNode("old", oldObj)
	displayNode("new", newObj)
}

func OnNodeDelete(obj interface{}) {
	displayNode("delete", obj)
}

func displayNode(msg string, obj interface{}) {
	node := obj.(*corev1.Node)
	fmt.Printf("%s %s\n", msg, node.Name)
}
```
kubeconfig
```yaml
apiVersion: v1
kind: Config
clusters:
- cluster:
    server: http://127.0.0.1:10550
  name: default
contexts:
- context:
    cluster: default
    namespace: default
    user: default
  name: default
current-context: default
users:
- name: default
  user:

```

On watch-node's startup, there are some add node events, and  update node events for the node on which I execute watch-node. After that,  no matter what I do to all nodes with kubectl, the watch-node didn't receive any more node events.
```
root@edge11:~# ./watch-node 
I0104 02:48:42.877915  569106 main.go:39] "Starting node controller"
I0104 02:48:42.877963  569106 shared_informer.go:240] Waiting for caches to sync for node

I0104 02:48:52.888388  569106 trace.go:205] Trace[1572303544]: "Reflector ListAndWatch" name:pkg/mod/k8s.io/client-go@v0.22.5/tools/cache/reflector.go:167 (04-Jan-2024 02:48:42.877) (total time: 10010ms):
Trace[1572303544]: ---"Objects listed" 10010ms (02:48:52.888)
Trace[1572303544]: [10.010531917s] [10.010531917s] END
add edge11
map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:edge11 kubernetes.io/os:linux node-role.kubernetes.io/agent: node-role.kubernetes.io/edge:]
I0104 02:48:52.979609  569106 shared_informer.go:247] Caches are synced for node 
old edge11
map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:edge11 kubernetes.io/os:linux node-role.kubernetes.io/agent: node-role.kubernetes.io/edge:]
new edge11
map[beta.kubernetes.io/arch:amd64 beta.kubernetes.io/os:linux kubernetes.io/arch:amd64 kubernetes.io/hostname:edge11 kubernetes.io/os:linux node-role.kubernetes.io/agent: node-role.kubernetes.io/edge:]


```

**What you expected to happen**:

metaServer notifies events for cluster resources

**How to reproduce it (as minimally and precisely as possible)**:
1. Execute watch-node on any edge node ( metaServer need to be enabled)
2. Change any nodes with kubectl, such as:
````
kubectl label node beijing test=bluven
kubectl label node beijing test-
``` 
3. Check the output of watch-node.

**Anything else we need to know?**:

After more expirement, It seems delete event is ok, but add/update are not provided.

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:38:33Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.5", GitCommit:"5c99e2ac2ff9a3c549d9ca665e7bc05a3e18f07e", GitTreeState:"clean", BuildDate:"2021-12-16T08:32:32Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
```
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`): 1.12.2
- <details><summary>Cloud nodes Environment:</summary>

  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>
- <details><summary>Edge nodes Environment:</summary>

  - edgecore version (e.g. `edgecore --version`):
  - Hardware configuration (e.g. `lscpu`):
  - OS (e.g. `cat /etc/os-release`):
  - Kernel (e.g. `uname -a`):
  - Go version (e.g. `go version`):
  - Others:

  </details>

**Comment by luomengY** [comment]
Do you enable metaserver on edge nodes? You can present your edgecoe.yaml.

**Comment by bluven** [comment]
@luomengY Of coure I eanbled metaServer and I mentioned it in the issue, otherwise the program wouldn't start at all.  Why won't you just run this program. 

```
apiVersion: edgecore.config.kubeedge.io/v1alpha2
database:
  aliasName: default
  dataSource: /var/lib/kubeedge/edgecore.db
  driverName: sqlite3
kind: EdgeCore
modules:
  dbTest:
    enable: false
  deviceTwin:
    enable: true
  edgeHub:
    enable: true
    heartbeat: 15
    httpServer: https://10.22.53.114:10002
    messageBurst: 60
    messageQPS: 30
    projectID: e632aba927ea4ac2b575ec1603d56f10
    quic:
      enable: false
      handshakeTimeout: 30
      readDeadline: 15
      server: 192.168.6.4:10001
      writeDeadline: 15
    rotateCertificates: true
    tlsCaFile: /etc/kubeedge/ca/rootCA.crt
    tlsCertFile: /etc/kubeedge/certs/server.crt
    tlsPrivateKeyFile: /etc/kubeedge/certs/server.key
    token: c59937a04ce2ecb903507f265cd03a34c1b6a34206354c6862118cb0129e18f0.eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDM2NDc5OTJ9.VahS0vqwpFWZvijIz2CXdG_4px6QurS7PhghHawO_Ug
    websocket:
      enable: true
      handshakeTimeout: 30
      readDeadline: 15
      server: 10.22.53.114:10000
      writeDeadline: 15
  edgeStream:
    enable: true
    handshakeTimeout: 30
    readDeadline: 15
    server: 10.22.53.114:10004
    tlsTunnelCAFile: /etc/kubeedge/ca/rootCA.crt
    tlsTunnelCertFile: /etc/kubeedge/certs/server.crt
    tlsTunnelPrivateKeyFile: /etc/kubeedge/certs/server.key
    writeDeadline: 15
  edged:
    cniBinDir: /opt/cni/bin
    cniCacheDir: /var/lib/cni/cache
    cniConfDir: /etc/cni/net.d
    containerRuntime: docker
    dockerEndpoint: unix:///var/run/docker.sock
    dockershimRootDirectory: /var/lib/dockershim
    enable: true
    hostnameOverride: edge11
    imagePullProgressDeadline: 1m0s
    kubeConfig: /etc/kubeedge/config/kubeconfig
    masterServiceNamespace: default
    maxContainerCount: -1
    maxPerPodContainerCount: 1
    minimumGCAge: 0s
    networkPluginMTU: 1500
    networkPluginName: cni
    nodeIP: 192.168.6.4
    nonMasqueradeCidr: 10.0.0.0/8
    podSandboxImage: kubeedge/pause:3.1
    registerNode: true
    registerNodeNamespace: default
    registerSchedulable: true
    remoteImageEndpoint: unix:///var/run/dockershim.sock
    remoteRuntimeEndpoint: unix:///var/run/dockershim.sock
    rootDirectory: /var/lib/edged
    seccompProfileRoot: /var/lib/edged/seccomp
    tailoredKubeletConfig:
      address: 127.0.0.1
      cgroupDriver: systemd
      cgroupsPerQOS: true
      clusterDNS:
      - 169.254.25.10
      clusterDomain: cluster.local
      configMapAndSecretChangeDetectionStrategy: Get
      containerLogMaxFiles: 5
      containerLogMaxSize: 10Mi
      contentType: application/json
      cpuCFSQuota: true
      cpuCFSQuotaPeriod: 100ms
      cpuManagerPolicy: none
      cpuManagerReconcilePeriod: 10s
      enableControllerAttachDetach: true
      enableDebugFlagsHandler: true
      enableDebuggingHandlers: true
      enableProfilingHandler: true
      enableServer: false
      enableSystemLogHandler: true
      enforceNodeAllocatable:
      - pods
      eventBurst: 10
      eventRecordQPS: 5
      evictionHard:
        imagefs.available: 15%
        memory.available: 100Mi
        nodefs.available: 10%
        nodefs.inodesFree: 5%
      evictionPressureTransitionPeriod: 5m0s
      failSwapOn: false
      fileCheckFrequency: 20s
      hairpinMode: promiscuous-bridge
      healthzBindAddress: 127.0.0.1
      healthzPort: 10248
      httpCheckFrequency: 20s
      imageGCHighThresholdPercent: 80
      imageGCLowThresholdPercent: 40
      imageMinimumGCAge: 2m0s
      iptablesDropBit: 15
      iptablesMasqueradeBit: 14
      kubeAPIBurst: 10
      kubeAPIQPS: 5
      logging:
        format: text
      makeIPTablesUtilChains: true
      maxOpenFiles: 1000000
      maxPods: 110
      memoryManagerPolicy: None
      memorySwap: {}
      memoryThrottlingFactor: 0.8
      nodeLeaseDurationSeconds: 40
      nodeStatusMaxImages: 50
      nodeStatusReportFrequency: 5m0s
      nodeStatusUpdateFrequency: 10s
      oomScoreAdj: -999
      podPidsLimit: -1
      readOnlyPort: 10350
      registryBurst: 10
      registryPullQPS: 5
      resolvConf: /etc/resolv.conf
      runtimeRequestTimeout: 2m0s
      seccompDefault: false
      serializeImagePulls: true
      shutdownGracePeriod: 0s
      shutdownGracePeriodCriticalPods: 0s
      streamingConnectionIdleTimeout: 4h0m0s
      syncFrequency: 1m0s
      topologyManagerPolicy: none
      topologyManagerScope: container
      volumePluginDir: /usr/libexec/kubernetes/kubelet-plugins/volume/exec/
      volumeStatsAggPeriod: 1m0s
  eventBus:
    enable: true
    eventBusTLS:
      enable: false
      tlsMqttCAFile: /etc/kubeedge/ca/rootCA.crt
      tlsMqttCertFile: /etc/kubeedge/certs/server.crt
      tlsMqttPrivateKeyFile: /etc/kubeedge/certs/server.key
    mqttMode: 2
    mqttPassword: ""
    mqttPubClientID: ""
    mqttQOS: 0
    mqttRetain: false
    mqttServerExternal: tcp://127.0.0.1:1883
    mqttServerInternal: tcp://127.0.0.1:1884
    mqttSessionQueueSize: 100
    mqttSubClientID: ""
    mqttUsername: ""
  metaManager:
    contextSendGroup: hub
    contextSendModule: websocket
    enable: true
    metaServer:
      enable: true
      server: 127.0.0.1:10550
      tlsCaFile: /etc/kubeedge/ca/rootCA.crt
      tlsCertFile: /etc/kubeedge/certs/server.crt
      tlsPrivateKeyFile: /etc/kubeedge/certs/server.key
    remoteQueryTimeout: 60
  serviceBus:
    enable: false
    port: 9060
    server: 127.0.0.1
    timeout: 60

```

**Comment by luomengY** [comment]
> ```go
> 24 * time.Hour
> ```

Your syncPeriod parameter is set to 24 * time Hour, do you understand the purpose of this parameter?

**Comment by Shelley-BaoYue** [comment]
@ZhengXinwei-F

**Comment by ZhengXinwei-F** [comment]
Thank you for your inquiry. Could you please give the MetaServer-related sections of the edgecore logs and the dynamiccontroller-related parts of the cloudcore logs?

**Comment by luomengY** [comment]
I tested and found no issues in kubedge v1.15.1, but this issue does exist in kubedge v1.13.0. Has this issue been fixed? I found the following errors in the cloudcore logs in v1.13.0:

**Comment by ZhengXinwei-F** [comment]
> @ZhengXinwei-F I tested and found no issues in kubedge v1.15.1, but this issue does exist in kubedge v1.13.0. Has this issue been fixed? I found the following errors in the cloudcore logs in v1.13.0: ![28415d462e1bf485828c491acf49e73](https://private-user-images.githubusercontent.com/110345347/295503207-a5bb9817-2e93-4494-908f-59a59b815109.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ4NzY3MTksIm5iZiI6MTcwNDg3NjQxOSwicGF0aCI6Ii8xMTAzNDUzNDcvMjk1NTAzMjA3LWE1YmI5ODE3LTJlOTMtNDQ5NC05MDhmLTU5YTU5YjgxNTEwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwMTEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDExMFQwODQ2NTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YTg0Zjk4OTRkOTAyZmJkN2ZlY2I4MTIzNWY4MDQxNDFkMGMyNjk5YmY2NWUzNjg4ZDFmNzQzNWI3MjE3ZjkzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KvQDMSjl8JFuUzrc3ZFZfDEZsTd7cAa3v1J_ovnkpGg)

This issue has been resolved in #4915.
@bluven  Is the case of your logs? To resolve the issue, you might consider upgrading to 1.14.
